### ISSUE TITLE
Errors during installation 

### ISSUE THREAD

**Issue by SchettlerKoehler** [devonfw/ide#466]
### Expected behavior

As a developer, I want devonfw ide to setup correctly so that I can use it without issues.

### Actual behavior
There are errors during the installation.

### Steps to reproduce

1. Download and extract devonfw-ide-scripts-2020.08.001.tar.gz
2. Run setup

### Comments/Hints:
Console log:
...
*** Setting up vscode ***
Starting installation of vscode to /root/devonfw/software/vscode from https://update.code.visualstudio.com/1.47.3/linux-x64/stable
Downloading vscode-1.47.3-linux.tgz from https://update.code.visualstudio.com/1.47.3/linux-x64/stable
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   130  100   130    0     0    122      0  0:00:01  0:00:01 --:--:--   122
100 85.3M  100 85.3M    0     0  43.8M      0  0:00:01  0:00:01 --:--:--  113M
Successfully extracted archive vscode-1.47.3-linux.tgz to updates/extracted
Successfully installed vscode
To be fully functional please rerun 'devon' command to update your PATH properly.
vsCode extensions not configured to be removed on update. If you would like to do so, please create /root/devonfw/settings/vscode/vscode.properties with clean_plugins_on_update=true.

******** ATTENTION ********
Execute plugin installation with following parameters ( --install-extension devonfw.devonfw-extension-pack) into VS code...
Please wait until VSCode has successfully installed all plugins.
You may then use or close VSCode as you like.
/root/devonfw/software/vscode/bin/../code: error while loading shared libraries: libXcursor.so.1: cannot open shared object file: No such file or directory
Failed to run command: '/root/devonfw/software/vscode/bin/code' --new-window --user-data-dir '/root/devonfw/conf/vscode' --extensions-dir '/root/devonfw/software/vscode-extensions' --force  --install-extension devonfw.devonfw-extension-pack

******** ATTENTION ********
ERROR: Failed to run command '/root/devonfw/software/vscode/bin/code'
We are sorry for the inconvenience. Please check the above errors, resolve them and try again.
Exit code was 127
ERROR: command 'vscode setup' failed with exit code 127

### Affected version:

* OS: Linux
* Browser: -

**Comment by markusschuh** [comment]
Your linux installation misses a prerequisite to run a UI program like vscode there. Have you installed packages for execution of a X11 compatible environment in your linux installation? (X.org, Wayland, ...) The details strongly depend on the linux distribution used. For any issue reports - also for other programs - I recommend to always add your linux distribution name and version, not just "linux" for OS.

linux is potentially one of the least used environments for `devonfw-ide`. At the same time in linux any script driven installation of UI programs is far more complicated than in macOS or Windows - just due to the diversity of different linux distributions, where many have their own package management systems. Of course, this diversity is also a strength of linux - but as often, an up side may have another down side.

Of course it were possible to add a dependency detection of needed prerequisites for the `devonfw-ide` components under linux to the setup - but this will need somebody, who has linux as his main development environment. I personally run a lot of build tasks on linux - but I never have seen that much need to add `devonfw-ide` there, since linux already has its own well suited means to create virtual environments of specific components. 

Your best option - in case you really need `devonfw-ide` to run under linux including vscode - may be to first setup vscode there using one of the methods supported by vscode itself at https://code.visualstudio.com/docs/setup/linux  Once a vscode as part of the linux operating system runs, all the needed prerequisites should be in place to also allow another vscode version to run, which is then installed by the `devonfw-ide` setup inside the software folder. 

If you may not intend to run vscode at all und linux and only want to have some other `devonfw-ide` features to be available under linux - e.g. the `devon build ...` - then you could exclude setup from installing it, e.g. by creation of a `devon.properties` within the `devonfw-ide`top dir before setup run, which sets the `DEVON_IDE_TOOLS` variable:

```
echo 'DEVON_IDE_TOOLS=(java mvn node npm ng)' >devon.properties
```

**Comment by hohwille** [comment]
@markusschuh thanks for your investigation. I came to the same conclusion and assume we can close this as invalid.
I labeld as `waiting-for-feedback` so lets give some time to wait. 
@SchettlerKoehler you could do as a favor and give feedback after reading the comment from markus. Thanks in advance.

**Comment by SchettlerKoehler** [comment]
Thank you for your feedback. I created the issue because it was working in the last version and has broken the existing katacoda tutorials which are using the default settings. We will provide a settings URL for the tutorials to fix that, but something that has been working is broken, now. We should be aware of that.

**Comment by markusschuh** [comment]
O.k. - so the runtime environment, where the issue appears is a katacode tutorial environment. You should mention this in new issues, since this is a very specific - and more complex than default - use case for running devonfw-ide.

Are you sure, that vscode has run before within that katacoda tutorial - or could it be, the vscode was excluded from being installed before? This can be done by settings the right variables before setup.

As https://www.katacoda.com/courses/vscode/playground shows, a vscode can be run within a katacode tutorial - but as the description "_early preview of VS Code Integration live on Katacoda.com_" implies, this is anything else then the default vscode archive for linux, that is installed by devonfw-ide. E.g. you can visit that vscode katacoda tutorial, start a "Terminal" from menu, check the running processes using `ps -ef` and finally list the shared libraries, this specific vscode uses:

```
$ ldd  /opt/katacoda-editor-vscode0.1-linux-x64 
        linux-vdso.so.1
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1
        libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6
        /lib64/ld-linux-x86-64.so.2
```

This is very different from the set of shared libraries - including libXcursor.so.1 - the the default vscode needs for proper run. Katacode does not provide an X11 runtime environment as far as I can say. That is the reason they need to work on their own forked vscode version - which looks quite nice, by the way.

So either remove vscode from install - if you don't need it. Or if you need it inside the katacode tutorial, try to replace the default vscode by the katacode one. But even then there is a high risk, that not all those vscode plugins - that devonfw-ide per default installs - would run also inside katacoda.

From viewpoint of devonfw-ide this is no bug, since katacoda can't be a supported runtime environment for the pure default set of installed software components. Consider it as a challenge to change that. Having a katacode tutorial for devonfw - including the vscode - sounds like a great idea to me.

**Comment by SchettlerKoehler** [comment]
I did not mention this to avoid confusion because in the katacoda tutorials you had two vscode instances. In the tutorials vscode of devon ide was installed to make the installation of devon ide easy for first time users (just press enter). The installed version was never used in the tutorials. The version used in the tutorials was always the one provided by katacoda. I opened the issue because the installation with the default settings was working and after the last release of devonfw it stopped working.
To fix the errors we added a SettingsURL without eclipse and vscode. This is OK for the tutorials. I just wanted to point out that the behavior has changed and that this is breaking the installation in some (edge) cases.
I think we can close the issue.

**Comment by markusschuh** [comment]
IMHO suppressing info in github issues more likely introduces mix ups than it helps to avoid them. But anyway ...

A katacoda tutorial for devon is without doubt a valuable step forward - so I'll try to help moving it out of the `edge`:
I find now https://www.katacoda.com/devonfw - sorry I missed this before.  So I've run now the Cobigen Cli tutorial - but used the 2020.04.004 release instead for installation.

```
wget -c 'https://repository.sonatype.org/service/local/artifact/maven/redirect?r=central-proxy&g=com.devonfw.tools.ide&a=devonfw-ide-scripts&p=tar.gz&v=2020.04.004' -O - | tar -xz
```

Here is the same `Exit code was 127` as result of `vscode setup` - but indeed `setup` continues. The different behaviour in 2020.04.004 vs. later versions is within the fix for #432  - the [code change](https://github.com/devonfw/ide/commit/326b4f6a564b40edcd2596f28a5b99f6dc5cbf53) introduced to run the plugin installation now in foreground - before it was a background job. So plugin installation errors do no longer stay undetected - as in older `devonfw-ide` releases.

So we agree, this issue can be closed. It is now clear, what caused the different behaviour inside the tutorial. And I hope we agree, that optimization of katacoda to only install components that need to be installed, is the best way forward.

Since `devonfw-ide` has enough configuration options - not only an own full new settings URL  - to control setup - I'll propose some enhancements within the https://github.com/devonfw/katacoda-scenarios that I now have detected.
